src/main/java/pl/piekoszek/gorskimatches/challange/ChallengeService.java

```diff
@@ -15,131 +15,152 @@ public class ChallengeService {
 
     private final EmailService emailService;
 
-    private final EquationRandomizer equationRandomizer;
-
-    private final GenerateUUID generateUUID;
-
     private final ChallengeRepository challengeRepository;
 
-    private TimeService timeService;
-
     private final Judge judge;
 
     private final String server;
 
+    private final QuizAnswerChecker quizAnswerChecker;
+
+    private final ChallengeGenerator challengeGenerator;
+
     public ChallengeService(EmailService emailService,
```

Comment:
More than 3 dependencies is good indicator that this class is doing too much. Please try to split it somehow.